Broad share spec

emattei opened the pull request:
SHARE-seq seqspec file for the fastqs that will be submitted by Broad.

sbooeshaghi reviewed (changes requested):
Thank you for taking the time submit a `seqspec` for the `SHARE-Seq` assay! I've added comments for my review. The `spec` has a good structure and a bit more work can make it entirely descriptive of the libraries generated with the SHARE-Seq assay.

The biggest point that needs addressing is the barcode structure. The `ATAC` region didn't contain any information about the `cell_barcodes`. This can be fixed by adding `Regions` in the `ATAC` modality that describe the barcode structure. Additionally, the barcode structure in the `RNA` modality appears to be missing important information about the combinatorial barcoding scheme that generates the final `cell_barcode`. This can be fixed by adding sub-`Regions` in the `cell_barcode` `Region`.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+- RNA
+
+assay_spec:
+    - region_id: ATAC
```
For `ATAC` fastq1 and fast2 only contain gDNA. The cell barcode is in the read name. That's why I asked in the IGVF slack if there was a way to describe that.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+# Assay region
+!Assay
+name: SHARE-Seq
+doi: 10.1016/j.cell.2020.09.056
```
Sure, I will change this.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+          region_id: R1.fastq.gz
+          region_type: R1.fastq.gz
+          name: Read 1 FASTQ
+          sequence_type: random
```
Can I just remove it and put it all in the parent `Region`?

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+          region_id: R2.fastq.gz
+          region_type: R2.fastq.gz
+          name: Read 2 FASTQ
+          sequence_type: random
```
I guess same thing here.  It seems that I don't have to have sub-`Regions`

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+- RNA
+
+assay_spec:
+    - region_id: ATAC
```
Oops sorry I missed that. Will fix

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+          region_id: R1.fastq.gz
+          region_type: R1.fastq.gz
+          name: Read 1 FASTQ
+          sequence_type: random
```
As before.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+          regions:
+          # Cell Barcode
+            - !Region
+              region_id: cell_barcode
```
We are not going to submit data straight out of the sequencer. What you can see in the synapse is the exact format that we will submit to IGVF and what I am trying to describe in this YAML.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+          regions:
+          # Cell Barcode
+            - !Region
+              region_id: cell_barcode
```
Same as before.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+              max_len: 24
+              order: 1
+              onlist: !Onlist
+                filename: "gs://broad-buenrostro-bcl-outputs/192_PLATE1_PLATE2_whitelist.txt"
```
Sounds good.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+      regions:
+        # Read 1 Fastq
+        - !Region
+          region_id: R1.fastq.gz
```
Will fix.

emattei commented on assays/SHARE-seq/broad_share_spec.yaml:
```diff
@@ -0,0 +1,112 @@
+              onlist: null
+        # Read 2 Fastq
+        - !Region
+          region_id: R2.fastq.gz
```
As above.

emattei commented:
* Added the barcode list
* Fixed doi url
* `region_id` is now unique
* Fixed `region_type` is now unique
* `!Region` now present
*  Fixed `sequence_type`
*  Changed barcode definitions

sbooeshaghi commented:
Hi Eugenio, thank you for making these changes. Could you please install the most recent version of seqspec and run `seqspec check` to verify the correctness and completeness of the spec? In addition, please supply FASTQ files (with 1 million reads) associated with the regions identified in the spec, in your PR. For more information please see the contributing page for more details (https://github.com/IGVF/seqspec/blob/main/docs/CONTRIBUTING.md). Thank you!

sbooeshaghi commented:
Hey Eugenio- just checking in. I'd like to get your spec merged into the repo!

emattei commented:
Hi Sina,
Since seqspec is going to be released per file I cannot provide a general version to put here anymore. What you have on the github is already enough as "general" seqspec for SHARE-seq. All the seqspecs relative to the files for IGVF will be in the portal. I guess we can close this pull request.
